### PR TITLE
feat: set admin_key during migration and unit tests

### LINKED ISSUE #14070: Set admin_key during migration and unit tests.

### PR DESCRIPTION
**Pull request by iwsimon**
**Description**:
Set admin_key during migration.
Added more unit tests to make 100% classes and lines covered under hedera-addressbook-service-impl.

**Related issue(s)**:

Fixes #14070 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by jeromy-cannon — APPROVED** [review]

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java [hunk 1]
```diff
@@ -63,6 +64,9 @@ public void migrate(@NonNull final MigrationContext ctx) {
                 ctx.newStates().get(NODES_KEY);
         final var networkInfo = ctx.networkInfo();
         final var addressBook = networkInfo.addressBook();
+        final var bootstrapConfig = ctx.configuration().getConfigData(BootstrapConfig.class);
+        final var adminKey =
+                Key.newBuilder().ed25519(bootstrapConfig.genesisPublicKey()).build();
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java:69]: Is this discussed. Setting genesis Key on all nodes ?

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java [hunk 2]
```diff
@@ -63,6 +64,9 @@ public void migrate(@NonNull final MigrationContext ctx) {
                 ctx.newStates().get(NODES_KEY);
         final var networkInfo = ctx.networkInfo();
         final var addressBook = networkInfo.addressBook();
+        final var bootstrapConfig = ctx.configuration().getConfigData(BootstrapConfig.class);
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java:67]: We should validate this works fine with @tinker-michaelj 's upgrade test

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java [hunk 3]
```diff
@@ -63,6 +64,9 @@ public void migrate(@NonNull final MigrationContext ctx) {
...
+        final var adminKey =
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java:68]: Are there E2E tests added for adding `adminKey` in `NodeCreate` and `NodeUpdateHandler`. Each PR should have E2E tests corresponding to the change

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java (hunk 1 above)
Comment (iwsimon) [reply hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java:69]: Requirements is to use Account 0.0.55's key as adminKey. From com.hedera.node.app.service.token.impl.schemas.SyntheticAccountCreator#generateSyntheticAccounts, you can see we use this key, superUserKey as 0.0.55 key.

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java (hunk 3 above)
Comment (iwsimon) [reply hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java:68]: We are going to work on E2E test after this.

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java (hunk 2 above)
Comment (iwsimon) [reply hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java:70]: Merged with the latest develop branch. Ran DabEnabledUpgradeTest and passed.

**Review by jeromy-cannon — APPROVED** [review]

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java [hunk 4]
```diff
@@ -78,14 +82,12 @@ public void migrate(@NonNull final MigrationContext ctx) {
...
                     .weight(nodeInfo.stake())
+                    .adminKey(adminKey)
                     .build();
             writableNodes.put(
                     EntityNumber.newBuilder().number(nodeInfo.nodeId()).build(), node);
         });
 
-        if (writableNodes.isModified()) {
-            ((WritableKVStateBase) writableNodes).commit();
-        }
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java:88]: Why is this deleted?

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM!Thanks @iwsimon

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java (hunk 1 above)
Comment (jsync-swirlds) [reply hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java:72]: Please forgive my possibly naive question.
Is this method called at genesis only, or is this possibly called long after genesis?  I ask because account `0.0.55` might _change keys_ after genesis, and (I would assume) might no longer match the value in bootstrap configuration.

I just wonder if we should read the actual key on the account rather than assuming it hasn't changed?

Comment (iwsimon) [reply hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java:72]: All system accounts use this superKey.

Comment (jsync-swirlds) [reply hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java:72]: And they never change it/cannot change it?

Comment (iwsimon) [reply hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java:72]: Add this to tomorrow's meeting topic. Not using the account here is because the account might not have been created at this point.

**Review by rbarkerSL — APPROVED** [review]
LGTM
